### PR TITLE
Fix obvious notice in Warehouse class

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.2.x
| Description?      | We use `Shop::getGroupFromShop` but, by default, it returns an id of the group while a line later we need information from the array, this PR fixes this obvious mistake. AH01071: Got error 'PHP message: PHP Warning:  Trying to access array offset on value of type int in classes/stock/Warehouse.php on line 329PHP message: PHP Warning:  Trying to access array offset on value of type int in. We also need to get shop group id correctly since once it's an object and once it's an array
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Just look at the code + auto tests
| UI Tests          | https://github.com/florine2623/testing_pr/actions/runs/12081102374 ✅ 
| Fixed issue or discussion?     | n/a
| Related PRs       | n/a
| Sponsor company   | PrestaShop SA

### CODE REVIEW

**Review by Hlavtox — DISMISSED** [review]

**Review by Hlavtox — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
